src/main/kotlin/com/terraformation/backend/accelerator/db/ParticipantProjectSpeciesStore.kt

```diff
@@ -1,19 +1,53 @@
 package com.terraformation.backend.accelerator.db
 
 import com.terraformation.backend.accelerator.model.ExistingParticipantProjectSpeciesModel
+import com.terraformation.backend.accelerator.model.NewParticipantProjectSpeciesModel
 import com.terraformation.backend.accelerator.model.ParticipantProjectSpeciesModel
+import com.terraformation.backend.accelerator.model.toModel
 import com.terraformation.backend.auth.currentUser
+import com.terraformation.backend.customer.model.requirePermissions
 import com.terraformation.backend.db.accelerator.ParticipantProjectSpeciesId
+import com.terraformation.backend.db.accelerator.tables.daos.ParticipantProjectSpeciesDao
+import com.terraformation.backend.db.accelerator.tables.pojos.ParticipantProjectSpeciesRow
 import com.terraformation.backend.db.accelerator.tables.references.PARTICIPANT_PROJECT_SPECIES
 import com.terraformation.backend.db.default_schema.ProjectId
+import com.terraformation.backend.db.default_schema.tables.daos.ProjectsDao
 import jakarta.inject.Named
 import org.jooq.Condition
 import org.jooq.DSLContext
 
 @Named
 class ParticipantProjectSpeciesStore(
     private val dslContext: DSLContext,
+    private val participantProjectSpeciesDao: ParticipantProjectSpeciesDao,
+    private val projectsDao: ProjectsDao,
 ) {
+  fun create(model: NewParticipantProjectSpeciesModel): ExistingParticipantProjectSpeciesModel {
+    requirePermissions { createParticipantProjectSpecies(model.projectId) }
+
+    // Participant project species can only be associated
+    // to projects that are associated to a participant
+    val project = projectsDao.fetchOneById(model.projectId)
```

Comment:
Does this seem like an alright place to put this? Or would this make more sense within a "service"? It seems alright to me since the project ID is a direct field on the entity.

Reply:
Yeah, I think this is fine. We don't want `Store` classes to depend on each other because of the risk of dependency cycles, but it's fine for a `Store` to depend on multiple `Dao` classes.

Another option (but to be clear: what you're doing here is fine) is to add a `getParticipantId(projectId)` method to `ParentStore`.